contracts/allowlist/ValidatorSmartContractAllowList.sol

```diff
@@ -0,0 +1,165 @@
+        require(allowedAccounts[account].allowed == false, "account to add is already on the allow list");
+
+        for (uint i=0; i < currentVotes[account].length; i++) {
+            require(currentVotes[account][i] != msg.sender, "sender has already voted to add account");
+        }
+        currentVotes[account].push(msg.sender);
+        emit Vote(account, msg.sender, currentVotes[account].length, numAllowedAccounts/2 + 1, true, false);
+    }
+
+    function voteToRemoveAccountFromAllowList(address account) external senderIsAllowed {
+        require(account != address(0), "account to be added cannot be 0");
+        require(allowedAccounts[account].allowed == true, "account to remove is not on the allow list");
+
+        for (uint i=0; i < currentVotes[account].length; i++) {
+            require(currentVotes[account][i] != msg.sender, "sender has already voted to remove account");
+        }
+        currentVotes[account].push(msg.sender);
+        emit Vote(account, msg.sender, currentVotes[account].length, numAllowedAccounts/2 + 1, false, false);
+    }
+
+    function removeVoteForAccount(address account) external senderIsAllowed {
+        for (uint i=0; i < currentVotes[account].length; i++) {
+            if (currentVotes[account][i] == msg.sender) {
+                currentVotes[account][i] = currentVotes[account][currentVotes[account].length-1];
+                currentVotes[account].pop();
+                break;
+            }
+        }
+        emit Vote(account, msg.sender, currentVotes[account].length, numAllowedAccounts/2 + 1, !(allowedAccounts[account].allowed), false);
+    }
+
+    function countVotes(address account) external senderIsAllowed returns(uint numVotes, uint requiredVotes, bool electionSucceeded) {
+        for (uint i=0; i < currentVotes[account].length; i++) {
+            if (allowedAccounts[currentVotes[account][i]].allowed) {
+                // only increment numVotes if account that voted is still allowed
+                numVotes++;
+            }
+        }
+        if (numVotes > numAllowedAccounts / 2) {
+            delete(currentVotes[account]);
+            if (allowedAccounts[account].allowed) {
+                numAllowedAccounts--;
+                if(allowedAccounts[account].activeValidator) {
+                    require(validators.length > 1, "cannot remove allowed account with last active validator");
+                    validators[allowedAccounts[account].validatorIndex] = validators[validators.length - 1];
+                    validators.pop();
+                }
+                delete(allowedAccounts[account]);
+            } else {
+                numAllowedAccounts++;
+                allowedAccounts[account] = accountInfo(true, false, 0);
+            }
+            emit AllowedAccount(account, allowedAccounts[account].allowed);
+        }
+        emit CountVotes(account, msg.sender, numVotes, numAllowedAccounts / 2 + 1, !(allowedAccounts[account].allowed));
```

Comment:
What is the idea behind emitting `CountVotes`? If counting the vote enacts a change, we already emit `AllowedAccount`. And otherwise, is it of interest to anybody but the caller (who can always query the result using the hash of the just submitted transaction)?

But if you see a use, please keep the event.